### PR TITLE
Discord bot: Handle dropped participants

### PR DESCRIPTION
**Pull request by rshipp**
### Description

During discord salesforce sync, when we encounter a dropped participant, revoke their invite and kick them from the server.

### Motivation and Context

Task: https://app.asana.com/0/1174274412967132/1200526976497661/f

### Screenshots

In the Audit Log after marking a participant as Dropped and running the sync command: 

Screen Shot 2021-08-04 at 12 33 13 PM

In the bot debug logs:

```
D, [2021-08-04T17:30:53.578864 #38] DEBUG -- : Processing participant
D, [2021-08-04T17:30:53.579080 #38] DEBUG -- : Found existing invite for participant <REMOVED>
D, [2021-08-04T17:30:53.579747 #38] DEBUG -- : Revoking dropped participant's invite
D, [2021-08-04T17:30:53.911131 #38] DEBUG -- : Found Discord User ID, re-configuring in case roles changed
D, [2021-08-04T17:30:53.911744 #38] DEBUG -- : Kicking dropped participant
```

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [ ] My code follows the code style of this project.
- [ ] I have added neccessary labels
- [ ] My PR title follows the convention in the guide
- [ ] I have assigned myself to the PR
- [ ] I have requested reviews from team members
- [ ] I have created a task for reviewers on PM tool
- [ ] All new and existing tests passed.

### CODE REVIEW

File: lib/discord_bot.rb [hunk 1]
```diff
@@ -877,16 +877,33 @@ def sync_salesforce_participant(server_id, participant)
...
+              # Probably another instance of the bot got to it first.
+              LOGGER.debug "Invite already deleted"
+              Honeycomb.add_field('sync_salesforce_participant.invite_deleted', false)
+            end
+          end
+        end
+
         # If they already had a code, they may have already signed up too;
         # check their contact record for a Discord User ID.
-        # TODO: if this is returned by apex query now, don't fetch contact yet.
```
Comment (rshipp) [inline lib/discord_bot.rb:889]: fixed this in an earlier pr but forgot to remove the todo

File: lib/discord_bot.rb [hunk 2]
```diff
@@ -877,16 +877,33 @@ def sync_salesforce_participant(server_id, participant)
...
+            begin
+              invite.delete("Participant record was marked as Dropped.")
+              Honeycomb.add_field('sync_salesforce_participant.invite_deleted', true)
+            rescue Discordrb::Errors::UnknownInvite
+              # Probably another instance of the bot got to it first.
+              LOGGER.debug "Invite already deleted"
+              Honeycomb.add_field('sync_salesforce_participant.invite_deleted', false)
+            end
+          end
+        end
```
Comment (rshipp) [inline lib/discord_bot.rb:903]: should refactor invite deletion at some point but I don't want to do it right now

Comment (sadleb) [reply lib/discord_bot.rb:903]: Yeah, this is basically a copy / paste of the code on L933. I'm fine to refactor in a future pass.

**Review by sadleb — APPROVED** [review]
Could be cleaner, but whatevs. ROI to refactor isn't high at the moment.

**Pull request merged**